scripts/smithy.lic

```diff
@@ -0,0 +1,139 @@
+
+  def self.determine_service_type_from(arg)
+    SERVICE_TYPES.find { |t| t.start_with? arg.downcase }
+  end
+
+  BLACKSMITH_DESCRIPTIONS = [
+    "old dwarven blacksmith",
+  ]
+
+  def self.blacksmith
+    GameObj.npcs.find do |npc|
+      BLACKSMITH_DESCRIPTIONS.include? npc.name
+    end
+  end
+
+  def self.print_usage
+    respond
+    respond "  Usage:"
+    respond
+    respond "  #{$clean_lich_char}#{Script.current.name} <count> <crit, damage, or sighting>"
+    respond
+  end
+
+  def self.perform_services(opts)
+    opts.fetch(:count).times do
+      perform_service_request(opts.fetch(:service))
+      perform_service_confirmation
+    end
+  end
+
+  def self.perform_service_request(service_type)
+    if result = dothistimeout("ask ##{blacksmith.id} about #{service_type}", 5, /Modified \w+ Type: #{service_type}/i)
+      result
+    else
+      respond
+      respond "  ERROR: doesn't seem that we were able to ask the blacksmith for work"
+      respond
+      exit
+    end
+  end
+
+  def self.perform_service_confirmation
+    if result = dothistimeout("ask ##{blacksmith.id} about confirm", 5, /quickly returns, idly polishing it with a dirty rag as it hands it back to you/)
+      result
+    else
+      respond
+      respond "  ERROR: doesn't seem that we were able to confirm our service request with the blacksmith"
+      respond
+      exit
+    end
+  end
+
+  def self.validate_script_arguments
+    unless Script.current.vars[2]
+      print_usage
+      exit
+    end
+
+    unless Script.current.vars[1] =~ /\A\d+\Z/
+      respond
+      respond "  Didn't recognize #{Script.current.vars[1]} as a valid number of services"
+      print_usage
+      exit
+    end
+
+    unless determine_service_type_from(Script.current.vars[2])
+      respond
+      respond "  Didn't recognize #{Script.current.vars[2]} as a valid service type"
+      print_usage
+      exit
+    end
+
+    Script.current.vars
+  end
+
+  def self.validate_blacksmith_presence
+    if blacksmith
+      blacksmith
+    else
+      respond
+      respond "  ERROR: I don't see a blacksmith in this list of npcs:"
+      respond
+      respond "  #{GameObj.npcs.map(&:name).join(", ")}"
+      respond
+      exit
+    end
+  end
+
+
+  def self.validate
+    validate_script_arguments && validate_blacksmith_presence
+  end
+
+  def self.normalize_input(count_input, service_input)
+    {}.tap do |opts|
+      opts[:count] = count_input.to_i
+      opts[:service] = determine_service_type_from service_input
+    end
+  end
+
+  def self.run
+    if validate
+      opts = normalize_input(*Script.current.vars[1..-1])
+
+      respond
+      respond "  Unpause #{$clean_lich_char}#{Script.current.name} to request:"
+      respond
+      respond "  #{opts[:count]} #{opts[:service]} services from #{blacksmith.name}"
+      respond
+      pause_script
+
+      perform_services(opts)
+    end
+  end
+end
+
+if defined? Lich
+  Smithy.run
```

Comment:
I originally snuck this in at the end of the class definition without the Lich guard like
```
module Smithy
  ...

  run
end
```

but I thought that was kind of unintuitive.